api/filter/http/ext_authz.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package envoy.api.v2.filter.http;
+
+import "google/protobuf/duration.proto";
+
+import "validate/validate.proto";
+
+// [#not-implemented-hide:]
+// External Authorization filter calls out to an external service over the
+// gRPC Authorization API defined by api/external_auth.proto.
+// A failed check will cause this filter to return 403 Forbidden.
+message ExtAuthz {
+
+  // The name of the upstream authorization cluster.
+  string cluster_name = 1 [(validate.rules).string.min_bytes = 1];
+
+  // The timeout in milliseconds for the authorization service RPC. If not
+  // set, this defaults to 20ms.
+  google.protobuf.Duration timeout = 2;
+
+  // The filter's behaviour in case the external authorization service does
+  // not respond back. If set to true then in case of failure to get a
+  // response back from the authorization service allow the traffic.
+  // Defaults to false.
+  bool failure_mode_allow = 3;
```

Comment:
If you timeout and failure mode, you surely want to specify retries as well. These things almost always go together. If you want to fail closed, you almost always need to retry 2 times.

Reply:
Let's keep this simple to begin with and leave off retries.

We can add retry function to the filter as a later increment.

Reply:
+1 let's keep it simple for the first iteration.